es/es-server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -188,11 +188,7 @@
     private static final Setting<Boolean> INDEX_MAPPING_SINGLE_TYPE_SETTING; // private - should not be registered
     static {
         Function<Settings, String> defValue = settings -> {
-            boolean singleType = true;
-            if (settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, null) != null) {
-                singleType = Version.indexCreated(settings).onOrAfter(Version.ES_V_6_1_4);
-            }
-            return Boolean.valueOf(singleType).toString();
+            return Boolean.valueOf(true).toString();
```

Comment:
function can be removed completely, just use `true` for the  `boolSetting(...)` below instead